File: Ejercicio 9/pagos.java
abstract class MetodoPago {
    public abstract String procesar(double monto);
}

class Tarjeta extends MetodoPago {
    public String procesar(double monto) {
        return "Pago de Tarjeta realizado por " + monto;
    }
}

class Nequi extends MetodoPago {
    public String procesar(double monto) {
        return "Pago por Transferencia realizado por " + monto;
    }
}

class Efectivo extends MetodoPago {
    public String procesar(double monto) {
        return "Pago en efectivo realizado por " + monto;
    }
}

public class Main {
    public static void main(String[] args) {
        MetodoPago[] pagos = {
            new Tarjeta(),
            new Nequi(),
            new Efectivo()
        };

        for (MetodoPago p : pagos) {
            System.out.println(p.procesar(100));
        }
    }
}

File: Ejercicio 9/pagos.py
from abc import ABC, abstractmethod

class MetodoPago(ABC):

    @abstractmethod
    def procesar(self, monto):
        pass


class Tarjeta(MetodoPago):
    def procesar(self, monto):
        return f"-Pago de Tarjeta realizado por {monto}"


class Nequi(MetodoPago):
    def procesar(self, monto):
        return f"-Pago por Transferencia realizado por {monto}"


class Efectivo(MetodoPago):
    def procesar(self, monto):
        return f"-Pago en efectivo realizado por {monto}"



pagos = [
    Tarjeta(),
    Nequi(),
    Efectivo()
]

for p in pagos:
    print(p.procesar(100))
